go-controller/pkg/node/secondary_node_network_controller.go

```diff
@@ -38,11 +43,28 @@ func NewSecondaryNodeNetworkController(cnnci *CommonNodeNetworkControllerInfo, n
 func (nc *SecondaryNodeNetworkController) Start(ctx context.Context) error {
 	klog.Infof("Start secondary node network controller of network %s", nc.GetNetworkName())
 
-	handler, err := nc.watchPodsDPU()
-	if err != nil {
-		return err
+	if config.OVNKubernetesFeature.EnableMultiNetwork && config.OvnKubeNode.Mode == types.NodeModeDPU {
```

Comment:
how come you dont have your || util.IsNetworkSegmentationSupportEnabled() check here?

Dont we need to wire the pod on primary network to DPU as well with UDN?

Reply:
hmm good question should I support DPU mode with UDNs? we don't have any tests for these bits so I went with the assumption of supporting DPU only with secondary networks (enablemultiplenetwork) and not with primarynetworks(enablenetseg)..

Reply:
I think you should update code like this. Things that are easy if statements where DPU should also be affected. That doesn't mean we claim full support for DPU yet, but its better than not having the proper conditional and then later someone wondering why DPU is excluded.

Reply:
ack I can update this to be enabled for dpu mode as well

Reply:
I have updated this to be `	if (config.OVNKubernetesFeature.EnableMultiNetwork || util.IsNetworkSegmentationSupportEnabled()) && config.OvnKubeNode.Mode == types.NodeModeDPU {`